src/UKI.jl

```diff
@@ -0,0 +1,242 @@
+module UKI
+
+using LinearAlgebra
+using DocStringExtensions
+
+export UKIObj
+export update_ensemble!
+"""
+    KIObj{FT<:AbstractFloat, IT<:Int}
+
+Struct that is used in Unscented Kalman Inversion (UKI)
+
+#Fields
+$(DocStringExtensions.FIELDS)
+"""
+struct UKIObj{FT<:AbstractFloat, IT<:Int}
+    "vector of parameter names (never used)"
+     unames::Vector{String}
+    "a vector of arrays of size N_ensemble x N_parameters containing the mean of the parameters (in each UKI iteration a new array of mean is added)"
+     θ_bar::Vector{Array{FT}}
```

Comment:
The descriptor says that the size of each vector element is N_ensemble x N_parameters, but `θ_bar::Vector{Array{FT}}` is a vector of 1D arrays.

Reply:
It might also be useful for integration with EKI and EKS if all these objects have the same attributes. Here you are storing the mean and covariance, whereas in `EKI.jl` we store all the parameters. Storing the parameters is useful when considering the emulation stage.